functions/ts/src/helm_template.ts

```diff
@@ -17,82 +17,214 @@
+class HelmTemplateError extends Error {
+  constructor(m: string) {
+    super(m);
 
-  let error;
-  try {
-    const child = spawnSync('helm', args);
-    error = child.stderr;
-    let objects = safeLoadAll(child.stdout);
-    objects = objects.filter((o) => isKubernetesObject(o));
-    configs.insert(...objects);
-  } catch (err) {
-    configs.addResults(generalResult(err, 'error'));
-  }
-  if (error && error.length > 0) {
-    configs.addResults(
-      generalResult(
-        `Helm template command results in error: ${error.toString()}`,
-        'error'
-      )
-    );
+    // Set the prototype explicitly.
+    Object.setPrototypeOf(this, HelmTemplateError.prototype);
   }
 }
 
-function readArguments(configs: Configs) {
-  const args: string[] = [];
-  let nameArg: string | undefined;
-  let pathArg: string | undefined;
-  const configMap = configs.getFunctionConfigMap();
-  if (!configMap) {
-    return args;
+class HelmTemplate {
+  configs: Configs;
+  data: Map<string, string> = new Map<string, string>();
+
+  constructor(configs: Configs) {
+    this.configs = configs;
+
+    this.getData();
   }
-  configMap.forEach((value: string, key: string) => {
-    if (key === CHART_NAME) {
-      nameArg = value;
-    } else if (key === CHART_PATH) {
-      pathArg = value;
-    } else {
+
+  getData() {
+    const data = this.configs.getFunctionConfigMap();
+    if (data !== undefined) {
+      this.data = data;
+    }
+    this.validData();
+  }
+
+  validData() {
+    // either local or remote
+    if (!this.data.has(CHART) && !this.data.has(LOCAL_CHART_PATH)) {
+      throw new Error(
+        `Either ${CHART} or ${LOCAL_CHART_PATH} needs to be provided.`
+      );
+    }
+    // cannot use both remote and local chart
+    if (this.data.has(CHART) && this.data.has(LOCAL_CHART_PATH)) {
+      throw new Error(
+        `Cannot use ${CHART} and ${LOCAL_CHART_PATH} at the same time.`
+      );
+    }
+    // CHART_REPO and CHART_REPO_URL are required for remote chart
+    if (
+      this.data.has(CHART) &&
+      (!this.data.has(CHART_REPO) || !this.data.has(CHART_REPO_URL))
+    ) {
+      throw new Error(
+        `${CHART_REPO} and ${CHART_REPO_URL} are required for remote chart`
+      );
+    }
+  }
+
+  // get arguments for helm template command
+  templateArgs(): string[] {
+    const args: string[] = [];
+    this.data.forEach((value: string, key: string) => {
+      // template flags should start with '-'
+      if (!key.startsWith('-')) {
+        return;
+      }
       args.push(key);
       args.push(value);
+    });
+
+    // Helm template expects name and chart path first so place those at the beginning
+    if (this.data.get(LOCAL_CHART_PATH) !== undefined) {
+      args.unshift(this.data.get(LOCAL_CHART_PATH)!);
+    }
+    if (this.data.get(CHART_NAME) !== undefined) {
+      args.unshift(this.data.get(CHART_NAME)!);
+    }
+
+    return args;
+  }
+
+  // run helm template command
+  async template() {
```

Comment:
Functions/method names should be a verb. e.g.:

runHelmTemplate

Reply:
Will do